### DISCUSSION TITLE
不是很能理解umi的这个配置式路由

### DISCUSSION THREAD

**Discussion by hellojoeyyoung** [umijs/umi#8901] [Q&A]
版本：3.5.0
问题：ant-design-pro项目配置routes
···JavaScript
{
    path: '/',
    name: '基础资料',
    exact: true,
    redirect: '/enterprise',
    indexRoute: './base/enterprise',
    component: './base/a',
    // layout: false,
    // https://pro.ant.design/zh-CN/docs/layout/#%E8%87%AA%E5%AE%9A%E4%B9%89%E5%B8%83%E5%B1%80
    routes: [
      {
        path: 'enterprise',
        name: '企业管理',
        exact: true,
        icon: '',
        component: './base/enterprise',
      },
      {
        path: '/entaccount',
        name: '账户管理',
        exact: true,
        icon: '',
        component: './base/entaccount',
      },
    ]
  },
···
结果：页面报错，Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports. Check the render method of `LoadableComponent`.

看大意是组件文件没有返回react组件，可是我确定已经返回了

**Comment by hellojoeyyoung** [comment]
// 以下是./base/enterprise.js

import React, {useEffect, useRef, useMemo} from 'react';
import { history, useModel, connect, useDispatch, useSelector, useStore } from "umi";
import {Popconfirm} from 'antd';
import ProTable from '@ant-design/pro-table';
// import request from 'umi-request';
import { remove, queryList } from '@/services/enterprise'
import dayjs from "dayjs";
import {fetchChildrenPermission} from "@/utils";

const Enterprise = () => {
  debugger
  // const model = useModel('enterprise')
  const state = useSelector(state => state)
  const store = useStore()
  const dispatch = useDispatch()

  const isAdmin = localStorage.getItem('token').split('.=')[1] === 'admin'

  const { initialState, loading, error, refresh, setInitialState } = useModel('@@initialState');

  const {showModify, showDelete} = useMemo(() => {
    const childrenPermission = fetchChildrenPermission('base_enterprise')({app: {user: initialState.currentUser}}) || []
    const showModify = childrenPermission.includes('modify')
    const showDelete = childrenPermission.includes('delete')

    return {
      showModify,
      showDelete
    }
  }, [initialState])


  useEffect(() => {

    // dispatch({
    //   type: 'enterprise/query',
    //   payload: {}
    // })
  })

  // https://procomponents.ant.design/components/schema/#valuetype
  const columns = [
    {
      title: '序号',
      dataIndex: 'index',
      valueType: 'indexBorder',
      width: 48,
    },
    {
      title: '企业类型',
      dataIndex: 'type',
      // copyable: true,
      ellipsis: true,
      tip: '标题过长会自动收缩',
      search: true,
      valueEnum: {
        '1': { text: '借款方' },
        '2': '供应商',
        '3': '资金方',
        '4': '物流企业',
        '*': '-'
      },
      formItemProps: {
        rules: [
          // {
          //   required: true,
          //   message: '此项为必填项',
          // },
        ],
      },
      renderFormItem: (_, {defaultRender, ...other}) => {
        return defaultRender(_);
      },
    },
    {
      disable: false,
      title: '企业编号',
      dataIndex: 'companyCode',
    },
    {
      disable: false,
      title: '企业简称',
      dataIndex: 'alias',
      ellipsis: true,
    },
    {
      disable: false,
      title: '最后修改人',
      dataIndex: 'modifyUser',
    },
    {
      disable: false,
      title: '最后修改时间',
      dataIndex: 'modifyTime',
      valueType: 'dateTime',
      render: (_, record) => dayjs(+record.modifyTime).format('YYYY-MM-DD HH:mm:ss'),
      search: {
        // transform: value =>
      }
    },
    {
      title: '操作',
      valueType: 'option',
      key: 'option',
      render: (text, record, _, action) => {
        return (
          <>
            {/*{showModify && record.type!==4 && !isAdmin && <a href='javascript:void(0)' onClick={() => onEditItem(record)} className="blue">修改</a>}*/}
            {showModify && showDelete && record.type!==4 && !isAdmin && <span className="dividing-line">|</span>}
            {showDelete && record.type!==4 &&<Popconfirm title='请确定是否删除？' okText='确定' cancelText='取消' onConfirm={() => onDeleteItem(record.id)}>
              <a className='blue' href='javascript:void(0)'>删除</a>
            </Popconfirm>}
          </>
        )
      }
    },
  ];

  let actionRef = useRef();

  const onDeleteItem = id => {
    remove({ id })
      .then(_ => {
        actionRef.current.reload()
      })
  }


  return (
    // <>
      <ProTable
        columns={columns}
        actionRef={actionRef}
        cardBordered
        request={(params = {}, sort, filter) => {
          console.log('sort & filter', sort, filter);
          const { current, ...other } = Object.assign({}, params, {
            ...history.location.query,
            current: history.location.query.pageNumber
          })
          const _params = {
            pageNumber: current,
            ...other
          }
          return queryList(_params)
            .then(res => {
              return {
                data: res.data.data,
                success: res.success,
                total: res.data.total
              }
            })
        }}
        debounceTime={200}
        editable={{type: 'multiple'}}
        columnsState={{
          persistenceKey: 'pro-table-singe-demos',
          persistenceType: 'localStorage',
          onChange(value) {
            console.log('value: ', value);
          },
        }}
        rowKey="id"
        search={{
          labelWidth: 'auto',
          searchText: '我的搜索',

        }}
        form={{
          // 由于search.transform配置了 transform，提交的参与与定义的不同这里需要转化一下
          // syncToUrl: (values, type) => {
          //   if (type === 'get') {
          //     return {
          //       ...values,
          //       created_at: [values.startTime, values.endTime],
          //     };
          //   }
          //   return values
          // },
          // https://github.com/ant-design/pro-components/issues/3923
          ignoreRules: false
        }}
        pagination={{
          onChange: (pageNumber, pageSize) => {
            // history.push({ pathname: history.location.pathname, query: {...history.location.query, pageSize, pageNumber }})
          },
          showSizeChanger: true,
          showQuickJumper: true,
          defaultPageSize: history.location.query.pageSize || 10,
          defaultCurrent: history.location.query.pageNumber || 1
        }}
        dateFormatter="string"
        headerTitle="列表"
        toolBarRender={() =>
          [
            //   <Button key="button" icon={<PlusOutlined/>} type="primary">
            //     新建
            //   </Button>,
            //   //@ts-expect-error
            //   <Dropdown key="menu" overlay={menu}>
            //     <Button>
            //       <EllipsisOutlined/>
            //     </Button>
            //   </Dropdown>,
          ]}
      />
    // </>
  )
}

// export default connect(({enterprise}) => ({enterprise}))(Enterprise)
export default Enterprise

**Comment by fz6m** [comment]
由于缺乏足够的信息，请提供 [最小重现](https://stackoverflow.com/help/minimal-reproducible-example) 的 demo 帮助排查问题。
